### PR TITLE
ObjC static functions

### LINKED ISSUE #500: C-style functions consuming Objective-C classes should allow the Objective-C class to be passed
A function declaration like:
```C
void foo(NSDate *date);
```

Generates this code:

```dart
void foo(ffi.Pointer<ObjCObject> date) {
  ...
}
```

Instead, it could generate code like:

```dart
void foo(NSDate date) {
  ...
}
```

Or at least there should be an option or variant that accepts the `NSDate`.

### PR DESCRIPTION
**Pull request by liamappelbe**
Use Dart wrapper objects when passing args or receiving results from ObjC static functions.

For ref counting, we assume functions do not return a retained reference by default (so we have to retain it ourselves). We also parse the function annotations, and if we see a `NS_RETURNS_RETAINED`, then the function is returning a retained reference, so we don't have to manually retain it in that case.

Fixes #500, #309

### CODE REVIEW

File: test/native_objc_test/static_func_test.m [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+- (void)dealloc {
+  if (counter != nil) --*counter;
+  [super dealloc];
+}
+
+@end
+
+StaticFuncTestObj* staticFuncOfObject(StaticFuncTestObj* a) {
```
Comment (dcharkes) [inline test/native_objc_test/static_func_test.m:42]: Can we move these (the actual functions used in the test) up, and the helper object down? Or does ObjC require an ordering?

File: test/native_objc_test/static_func_test.dart [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+      generateBindingsForCoverage('static_func');
+    });
+
+    doGC() {
+      final gcNow = "gc-now".toNativeUtf8();
+      executeInternalCommand(gcNow.cast(), nullptr);
+      calloc.free(gcNow);
+    }
+
+    Pointer<Int32> staticFuncOfObjectRefCountTest(Allocator alloc) {
```
Comment (dcharkes) [inline test/native_objc_test/static_func_test.dart:47]: It seems this only has one call site, is this because of scoping that it's in a separate function?

**Review by dcharkes — APPROVED** [review]
Lgtm!

Thanks @liamappelbe!

File: test/native_objc_test/static_func_test.m (hunk 1 above)
Comment (liamappelbe) [reply test/native_objc_test/static_func_test.m:18]: Done. Just need to have the interface decl above them I think.

File: test/native_objc_test/static_func_test.dart (hunk 2 above)
Comment (liamappelbe) [reply test/native_objc_test/static_func_test.dart:47]: Yep. I use the function scoping and `doGC()` to ensure that all the wrapper objects are collected.

**Pull request merged**